pkg/extensions/search/cve/trivy/scanner.go

```diff
@@ -174,54 +175,61 @@ func (scanner Scanner) runTrivy(opts flag.Options) (types.Report, error) {
 	return report, nil
 }
 
-func (scanner Scanner) IsImageFormatScannable(repo, tag string) (bool, error) {
-	image := repo + ":" + tag
+func (scanner Scanner) IsImageFormatScannable(repo, ref string) (bool, error) {
+	var (
+		digestStr = ref
+		mediaType string
+	)
 
-	if scanner.cache.Get(image) != nil {
-		return true, nil
-	}
+	if zcommon.IsTag(ref) {
+		imgDescriptor, err := repodb.GetImageDescriptor(scanner.RepoDB, repo, ref)
+		if err != nil {
+			return false, err
+		}
 
-	repoMeta, err := scanner.repoDB.GetRepoMeta(repo)
-	if err != nil {
-		return false, err
+		digestStr = imgDescriptor.Digest
+		mediaType = imgDescriptor.MediaType
+	} else {
+		var found bool
+
+		found, mediaType = repodb.FindMediaTypeForDigest(scanner.RepoDB, godigest.Digest(ref))
+		if !found {
+			return false, zerr.ErrManifestNotFound
+		}
 	}
 
-	var ok bool
+	return scanner.IsImageMediaScannable(repo, digestStr, mediaType)
+}
 
-	imageDescriptor, ok := repoMeta.Tags[tag]
-	if !ok {
-		return false, zerr.ErrTagMetaNotFound
-	}
+func (scanner Scanner) IsImageMediaScannable(repo, digestStr, mediaType string) (bool, error) {
+	image := repo + "@" + digestStr
 
-	switch imageDescriptor.MediaType {
+	switch mediaType {
 	case ispec.MediaTypeImageManifest:
-		ok, err := scanner.isManifestScanable(imageDescriptor)
+		ok, err := scanner.IsManifestScanable(digestStr)
 		if err != nil {
 			return ok, fmt.Errorf("image '%s' %w", image, err)
 		}
 
 		return ok, nil
 	case ispec.MediaTypeImageIndex:
-		ok, err := scanner.isIndexScanable(imageDescriptor)
+		ok, err := scanner.IsIndexScanable(digestStr)
 		if err != nil {
 			return ok, fmt.Errorf("image '%s' %w", image, err)
 		}
 
 		return ok, nil
+	default:
+		return false, nil
 	}
-
-	return false, nil
 }
 
-func (scanner Scanner) isManifestScanable(descriptor repodb.Descriptor) (bool, error) {
-	manifestDigestStr := descriptor.Digest
-
-	manifestDigest, err := godigest.Parse(manifestDigestStr)
-	if err != nil {
-		return false, err
+func (scanner Scanner) IsManifestScanable(digestStr string) (bool, error) {
```

Comment:
It becomes very difficult to test the functions if we make them unexported. We can't use the `mocks` package inside internal tests because it causes a import cycle.

Reply:
Is this because of the changes to add `TestVulnerableLayer` in the test package? Maybe that test should be part of the this package instead.